Sources/PT.PM.AntlrUtils/AntlrConverter.cs

```diff
@@ -311,13 +323,24 @@ protected Token ExtractLiteral(IToken token)
                     text = text.Substring(1, text.Length - 2);
                     result = new StringLiteral(text, textSpan);
                 }
-                else if (text.All(c => char.IsDigit(c)))
+                else if (text.All(char.IsDigit))
                 {
-                    result = new IntLiteral(long.Parse(text), textSpan);
+                    if (int.TryParse(text, out int intValue))
+                    {
+                        result = new IntLiteral(intValue, textSpan);
+                    }
+                    else if (long.TryParse(text, out long longValue))
+                    {
+                        result = new LongLiteral(longValue, textSpan);
+                    }
+                    else
+                    {
+                        result = new BigIntLiteral(BigInteger.Parse(text), textSpan);
+                    }
                 }
                 else if (text.StartsWith("0X", StringComparison.OrdinalIgnoreCase))
```

Comment:
What about "0x" prefix? Or `text` gets capitalized before this comparison?

Reply:
Take attention: the `StringComparison.OrdinalIgnoreCase` option is used here. Thus there is no need to lower input string.

Reply:
Sorry, my bad! :smile: